docs/ref/models/fields.txt

```diff
@@ -1793,6 +1793,16 @@ Field API reference
 
         where the arguments are interpolated from the field's ``__dict__``.
 
+    .. attribute:: descriptor_class
+
+       .. versionadded:: 3.0
+
+       A class implementing the :py:ref:`descriptor protocol <descriptors>`
+       that is instantiated and assigned to the model instance attribute. The
+       constructor must accept a single argument, the field instance.
+       Overriding this class attribute allows for customizing get and set
+       behavior.
```

Comment:
Please move `versionadded` below the attribute description
```
       Overriding this class attribute allows for customizing get and set
       behavior.

.. versionadded:: 3.0

    The ``descriptor_class`` class attribute was added.
```

Reply:
Ok, no problem.

FWIW, the contributor guide recommends putting `versionadded` above the text.

https://docs.djangoproject.com/en/2.2/intro/contributing/#writing-documentation

https://docs.djangoproject.com/en/2.2/internals/contributing/writing-documentation/

> If a function, attribute, etc. is added, it’s also okay to use a `versionadded` annotation like this:

```
.. attribute:: Author.middle_name

    .. versionadded:: A.B

    An author's middle name.
```

I'm fine either way, but thought I would point out why I chose this form. It is consistent with the docs.

Reply:
Agreed, in this case `versionadded` looks better below the attribute, we don't need a separate `versionadded` section :+1: .